elyra/pipeline/component_catalog.py

```diff
@@ -44,40 +48,144 @@
+                # Make the required update to the manifest file
+                if os.path.isfile(self._manifest_filename):
+                    with open(self._manifest_filename, 'r') as f:
+                        manifest = json.loads(f.read())
+
+                write_required = False
+                if action == 'cache':
+                    # A 'cache' action means that updates to the component cache are required
+
+                    # Handle all pending actions in one batch
+                    for pending_action in manifest.get('actions'):
+                        catalog_name = list(pending_action.keys())[0]
+                        cache_action = list(pending_action.values())[0]
+
+                        write_required = True
+
+                        catalog_instance: ComponentCatalogMetadata = None
+                        if cache_action == 'delete':
+                            # Remove status stanza from manifest
+                            manifest['status'].pop(catalog_name, None)
+
+                            # Fabricate a metadata instance that only includes catalog name
+                            catalog_instance = ComponentCatalogMetadata(name=catalog_name)
+
+                        elif cache_action == 'modify':
+                            # Create or replace the status stanza to indicate status is current
+                            errors = manifest['status'].get(catalog_name, {}).get('errors', [])
+                            manifest['status'][catalog_name] = {
+                                'action': 'current',
+                                'errors': errors
+                            }
+
+                            # Fetch the catalog instance associated with this action
+                            catalog_instance = MetadataManager(
+                                schemaspace=ComponentCatalogs.COMPONENT_CATALOGS_SCHEMASPACE_ID
+                            ).get(name=catalog_name)
+
+                        # Add action to the cache queue
+                        self._cache_queue.put((catalog_instance, cache_action))
+
+                    # Clear redundant actions from manifest queue and
+                    # reset manifest actions to prepare for write out
+                    self.clear_duplicate_cache_actions()
+                    manifest['actions'] = []
+
+                else:
+                    # Any action that is not a 'cache' action means that it is a manifest action
+                    # Format necessary action and include in manifest only if not already present
+                    manifest_action = {catalog.name: action}
+                    if manifest_action not in manifest['actions']:
+                        manifest['actions'].append(manifest_action)
+                        write_required = True
+
+                if write_required:
+                    with open(self._manifest_filename, 'w') as f:
+                        f.write(json.dumps(manifest, indent=2))
```

Comment:
Per previous comment, we should use `dump()`:
```suggestion
                        json.dump(manifest, indent=2, f)
```